### PR TITLE
[W-11239126] - Frequency for every 1st and 15th case

### PR DESCRIPTION
**Pull request by maurimart**
[Technical Design Document W-11239126](https://salesforce.quip.com/M5AVAWjhPxCC)

# Critical Changes

Add frequencyFirstAndFifteenth custom label for frequency field.
Add validation on 1st and 15th case on get frequency field function.

# Changes

### CODE REVIEW

**Comment by maurimart** [comment]
[W-11239126](https://gus.lightning.force.com/a07EE00000yOhoHYAS)

**Review by gerardorodriguezsf — APPROVED** [review]
LGTM

**Review by JoseRossiSF — APPROVED** [review]
LGTM

**Review by force2b — CHANGES_REQUESTED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 1]
```diff
@@ -6040,6 +6040,14 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <shortDescription>Frequency processed text</shortDescription>
         <value>Every {0} weeks</value>
     </labels>
+    <labels>
+        <fullName>frequencyFirstAndFifteenth</fullName>
```
[the remark this answers is not in the document]
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml]: Changed!!

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 2]
```diff
@@ -6040,6 +6040,14 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <shortDescription>Frequency processed text</shortDescription>
         <value>Every {0} weeks</value>
     </labels>
+    <labels>
+        <fullName>RD_frequencyFirstAndFifteenth</fullName>
```
Comment (force2b) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:6044]: I see you changed this _one_ label, but what about all the other `frequency*` labels above and below it? Consistency in labels in important. None of these should have been created without the RD prefix. Plus, these labels, which now all start with RD, should be moved in the label file to the RD section where they belong.

**Review by force2b — CHANGES_REQUESTED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 2 above)
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml]: prefixes were added and labels were relocated according to their category!

**Comment by gerardorodriguezsf** [comment]
> What about the badly named labels below the ones you updated? They're all part of this feature. It's not even clear what they're used for or why the values are things like "nd".
> 
> Why are these even categorized as "recurring donations"? What do they have to do with RD?
> 
> * firstOrdinalNumber
> * secondOrdinalNumber
> * thirdOrdinalNumber
> * otherOrdinalNumber
> * creditCardElevatePayment
> * achElevatePayment
> 
> They're ALL also in the wrong part of the file. We generally try to keep these alphabetized. All of these labels, including the ones you changed should be the rest of the labels that begin with "RD" in the fullName.

Hello @force2b, these labels are used by the Recurring Donations LWC:
the custom labels firstOrdinalNumber, secondOrdinalNumber, thirdOrdinalNumber and otherOrdinalNumber are used to create the processed frequency text used in the 'Recurring Donations' LWC in the Frequency column, they are added to the end of the frequency string to be 1st, 2nd, 3rd, 4th, etc. It was planted this way to make it easy to translate since not all languages ​​use ordinal numbers in dates.
The custom labels creditCardElevatePayment and achElevatePayment are used to create de processed Payment Method used in the 'Recurring Donations' LWC in the Payment Method column.
These custom labels were already reviewed by @lparrott in this [PR](https://github.com/SalesforceFoundation/NPSP/pull/6973) 

@maurimart Please keep these alphabetized, and add the RD prefix.

**Review by force2b — APPROVED** [review]

**Pull request merged**